meos/src/general/meos.c

```diff
@@ -0,0 +1,467 @@
+        ok = false;    /* conflicting styles */
+      newDateStyle = USE_GERMAN_DATES;
+      have_style = true;
+      /* GERMAN also sets DMY, unless explicitly overridden */
+      if (!have_order)
+        newDateOrder = DATEORDER_DMY;
+    }
+    else if (pg_strcasecmp(tok, "YMD") == 0)
+    {
+      if (have_order && newDateOrder != DATEORDER_YMD)
+        ok = false;    /* conflicting orders */
+      newDateOrder = DATEORDER_YMD;
+      have_order = true;
+    }
+    else if (pg_strcasecmp(tok, "DMY") == 0 ||
+         pg_strncasecmp(tok, "EURO", 4) == 0)
+    {
+      if (have_order && newDateOrder != DATEORDER_DMY)
+        ok = false;    /* conflicting orders */
+      newDateOrder = DATEORDER_DMY;
+      have_order = true;
+    }
+    else if (pg_strcasecmp(tok, "MDY") == 0 ||
+         pg_strcasecmp(tok, "US") == 0 ||
+         pg_strncasecmp(tok, "NONEURO", 7) == 0)
+    {
+      if (have_order && newDateOrder != DATEORDER_MDY)
+        ok = false;    /* conflicting orders */
+      newDateOrder = DATEORDER_MDY;
+      have_order = true;
+    }
+    else if (pg_strcasecmp(tok, "DEFAULT") == 0)
+    {
+      /*
+       * We take the default value from PostgreSQL "ISO, MDY" and
+       * recursively parse it.
+       *
+       * We can't simply "return check_datestyle(...)" because we need
+       * to handle constructs like "DEFAULT, ISO".
+       */
+      char *subval = "ISO, MDY";
+      void *subextra = NULL;
+      if (!check_datestyle(&subval, &subextra))
+      {
+        pfree(subval);
+        ok = false;
+        break;
+      }
+      myextra = (int *) subextra;
+      if (!have_style)
+        newDateStyle = myextra[0];
+      if (!have_order)
+        newDateOrder = myextra[1];
+      pfree(subval);
+      pfree(subextra);
+    }
+    else
+    {
+      meos_error(ERROR, MEOS_ERR_TEXT_INPUT,
+        "Unrecognized key word: \"%s\".", tok);
+      pfree(rawstring);
+      free_stringlist(&elemlist);
+      return false;
+    }
+  }
+
+  pfree(rawstring);
+  free_stringlist(&elemlist);
+
+  if (!ok)
+  {
+    meos_error(ERROR, MEOS_ERR_TEXT_INPUT,
+      "Conflicting \"datestyle\" specifications.");
+    return false;
+  }
+
+  /* Set up the "extra" struct actually used by assign_datestyle */
+  myextra = (int *) palloc(2 * sizeof(int));
+  if (!myextra)
+    return false;
+  
+  myextra[0] = newDateStyle;
+  myextra[1] = newDateOrder;
+  *extra = (void *) myextra;
+
+  return true;
+}
+
+/**
+ * @brief Set the DateStyle
+ */
+bool
+meos_set_datestyle(char *newval, void *extra)
+{
+  if (! check_datestyle(&newval, &extra))
+    return false;
+    
+  int *myextra = (int *) extra;
+  DateStyle = myextra[0];
+  DateOrder = myextra[1];
+  return true;
+}
+
+/**
+ * @brief Get the DateStyle
+ */
+char *
+meos_get_datestyle(void)
+{
+  char *result = palloc(32);
+  if (!result)
+    return false;
+  sprintf(result, "%s, %s", datestyle_string(DateStyle), 
+    dateorder_string(DateStyle));
+  return result;
+}

```

Comment:
This should probable raise an error or return `NULL` instead of returning `false` when `palloc` fails.